src/utils/subscribe-key.ts

```diff
@@ -0,0 +1,22 @@
+import { RevalidateCallback, StateUpdateCallback } from '../types'
+
+// Add a callback function to a list of keyed callback functions and return
+// the unsubscribe function.
+export const subscribeCallback = (
+  key: string,
+  callbacks: Record<string, (RevalidateCallback | StateUpdateCallback)[]>,
+  callback: RevalidateCallback | StateUpdateCallback
+) => {
```

Comment:
The type-checking will still pass if you remove `RevalidateCallback`
This function does not require any details about the callback. 
Using `(...args: any[]) => any` might be enough


```suggestion
type Callback = (...args: any[]) => any
export const subscribeCallback = (
  key: string,
  callbacks: Record<string, Callback[]>,
  callback: Callback
) => {
```